### PR TITLE
fix: ensure attributes with no value, or "slightly truthy values" pass .attr

### LINKED ISSUE #47: `attr` fails for falsey attribute values
Hey everyone, great work on the awesome chai plugin!

The `.attr()` assertion fails when asserting on an attribute with a falsey value, such as an empty string.

Consider the following html:

```html
<article itemscope itemtype="http://schema.org/Article" itemprop="article"></article>
```

Asserting `expect(article).to.have.attr('itemscope')` fails when using enzyme's `mount`. It fails because [ReactTestWrapper calls `getAttribute(name) || undefined`](https://github.com/producthunt/chai-enzyme/blob/master/src/ReactTestWrapper.js#L31-L33) - and the `itemscope`'s result from `getAttribute` is `''`.

It should be noted that (although I haven't actually tested) - I believe that the CheerioWrapper won't do the same ([it doesn't use `|| undefined` in the wrapper code](https://github.com/producthunt/chai-enzyme/blob/master/src/CheerioTestWrapper.js#L33-L35) and [Cheerio itself returns the property verbatum](https://github.com/cheeriojs/cheerio/blob/master/lib/api/attributes.js#L38-L41)). 

### Solution

Do not use `|| undefined`. Instead pass the attribute verbatim. `getAttribute` will return `null` for attributes which don't exist on the element, which can be used to accurately detect the lack of an attribute over falsey values.

### PR DESCRIPTION
**Pull request by keithamus**
This wraps the mount wrapper in a `hasAttribute` check and if the value is falsey, it returns
true anyway - this way JSX like `<div itemScope />` will return true to having the itemscope
attribute.

Fixes #47. Closes #66 (picks up from @tleunen's work and completes it per @ljharb's comments).

Ping for people involved in #47: @vesln @tleunen @ljharb 
Ping for people involved in #66: @vesln @tleunen @ljharb @ayrton

### CODE REVIEW

**Comment by ayrton** [comment]
👍

**Comment by ayrton** [comment]
Thanks Keith, and thank you @tleunen for you initial work on this

**Comment by vesln** [comment]
👍

File: test/attr.test.js [hunk 1]
```diff
@@ -30,6 +31,22 @@ describe('#attr', () => {
...
+    it('passes when attribute exists without a value', (wrapper) => {
+      expect(wrapper.find('article')).to.have.attr('itemscope')
+    })
+
+    it('passes when attribute exists with a falsey (but not false/null/undefined) value', (wrapper) => {
+      expect(wrapper.find('article')).to.have.attr('draggable')
+    })
+
+    it('passes when attribute exists but has value `false`', (wrapper) => {
+      expect(wrapper.find('article')).to.not.have.attr('hidden')
```
Comment (ljharb) [inline test/attr.test.js]: wait - if the value is false, it absolutely "has" the attribute. jsx attributes can have either boolean value, otherwise how would you pass `PropTypes.bool` and pass both `true` and `false`?

Comment (keithamus) [inline test/attr.test.js]: props aren't quite attributes though. The attribute isn't rendered to the final dom because its not a valid value for that attr.

Comment (keithamus) [inline test/attr.test.js]: Its all more complicated than these tests indicate, because each attribute has potential subtle differences. For example `hidden`, in React, behaves via boolean values (see [their code](https://github.com/facebook/react/blob/4338c8db4b03cb1ba5f2e229b1baa5a848fe65ab/src/renderers/dom/shared/HTMLDOMPropertyConfig.js#L79)) so `true` will render `hidden` where `false` wont. But for something like [`spellCheck`](https://github.com/facebook/react/blob/4338c8db4b03cb1ba5f2e229b1baa5a848fe65ab/src/renderers/dom/shared/HTMLDOMPropertyConfig.js#L140), passing `false` will render `spellcheck="false"` and true `spellcheck="true"`. There's lots of esoteric values for a [range of attributes](https://github.com/facebook/react/blob/4338c8db4b03cb1ba5f2e229b1baa5a848fe65ab/src/renderers/dom/shared/HTMLDOMPropertyConfig.js#L43-L53) - we could test them a subset of them individually if you'd be happier with that?

Comment (ljharb) [inline test/attr.test.js]: I'd much prefer having explicit tests for all the edge cases.

Comment (ayrton) [inline test/attr.test.js]: I agree it's probably best to have a 1:1 behaviour with the FB codebase - for now this is already a huge improvement, so imo leave a note in the codebase and 🚢 unless you have the extra time to tackle this.

Comment (keithamus) [inline test/attr.test.js]: I'll get on this and so some better tests at some point today or tomorrow.

Comment (keithamus) [inline test/attr.test.js]: @ljharb @ayrton I've rebased, added a bunch of tests including the HAS_BOOLEAN_VALUE, HAS_NUMERIC_VALUE and regular string attrs. Hopefully this covers all of the edge cases around attrs... there's quite a few s😓

**Comment by keithamus** [comment]
Just noticed this had a bunch of lint errors. Fixed them now. Build should pass. @ljharb @ayrton happy with the extra tests?

**Comment by ayrton** [comment]
This is absolutely epic @keithamus thanks for the amazing work.
